packages/dataviews/src/dataforms-layouts/data-form-layout.tsx

```diff
@@ -47,6 +49,34 @@ export function DataFormLayout< Item >( {
 		[ form ]
 	);
 
+	const firstValidationRunRef = useRef( false );
+
+	useEffect( () => {
```

Comment:
Why do we need to trigger validation in an effect and not as an event handler?

Reply:
This runs the validation when the component is mounted. In the case of the pattern modal, we want to disable the button if the form isn't valid:



https://github.com/user-attachments/assets/79823da1-adea-4257-9dd2-c9b820b722c3

The validation runs on the onChange event too:

https://github.com/WordPress/gutenberg/blob/03e934ba72b316380c02ec456034dad690f6f6a0/packages/dataviews/src/dataforms-layouts/data-form-layout.tsx#L123-L152

Reply:
Could we maybe also move this logic and trigger it in the `DataForm` component maybe?
The `DataFormLayout` is also being used within the panel layout, meaning every time the dropdown is opened this logic gets run again.

Reply:
I don't think you need (or should) disable the submit button. Clicking on it should highlight form errors. I think we looked into disabling it but [research from Baymard adivsed otherwise.](https://baymard.com/premium/guideline-collections/pma153/699)

I don't want to have strong opinions here but this seems to be the most accessible standard, see the examples from both WooCommerce and Shopify's Checkout:


https://github.com/user-attachments/assets/3734e65f-2176-46bf-bfde-6d59300cab2a


https://github.com/user-attachments/assets/7929f0b8-a8ac-42cc-b738-23088c15e611